src/util/timestamper.h

```diff
@@ -0,0 +1,82 @@
+/// \file timestamper.h
+/// \brief Emit timestamps
+/// \author Kareem Khazem <karkhaz@karkhaz.com>
+
+#ifndef CPROVER_UTIL_TIMESTAMPER_H
+#define CPROVER_UTIL_TIMESTAMPER_H
+
+#ifdef _WIN32
+#define OPT_TIMESTAMP ""
+#define HELP_TIMESTAMP ""
+#else
+#define OPT_TIMESTAMP "(timestamp):"
+
+#define HELP_TIMESTAMP \
+" --timestamp <monotonic|wall> print microsecond-precision timestamps.\n"  \
+"                              monotonic: timestamps are relative to an\n" \
+"                              arbitrary start point, guaranteed to\n"     \
+"                              increase, formatted as s.u where 's'\n"     \
+"                              increases every second and 'u' is a zero-\n"\
+"                              padded six-digit microsecond counter.\n"    \
+"                              wall: timestamps use the format\n"          \
+"                              %Y-%m-%dT%H:%M:%S.u where 'u' is a six-\n"  \
+"                              digit zero-padded microsecond counter.\n"   \
+"                              Timestamp may decrease on daylight saving\n"\
+"                              time changes\n"
+#endif
+
+
+#include <memory>
+#include <sstream>
+
+/// \brief Timestamp class hierarchy
+///
+/// This class hierarchy supports generation of timestamps in various
+/// textual formats. The timestamps returned by instances of this class
+/// are empty; more meaningful timestamps are returned by derived classes.
+class timestampert
```

Comment:
Clarify whether this abstract, or is instantiating the base useful / sensible?

Reply:
This is not abstract, it is instantiated when `--timestamps` is not passed on the command line. This means that ui_messaget can always call `time->stamp()` without bothering to check whether `--timestamps` was passed; virtual dispatch takes care of whether to emit an actual timestamp or just the empty string. Should I clarify this in comments?

Reply:
Yes please, it looks at the moment like an abstract base that provides a default implementation for a  concrete subclass